dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -649,6 +649,7 @@ int fetchNextRow(uchar* buf, cal_table_info& ti, cal_connection_info* ci, bool h
                             break;
 
                         default:
+                            (*f)->field_length = 255; // reserve enough space for varchar, was previously displaying only 21 characters
```

Comment:
After spending a great deal of time digging into CS and MDB trying to figure out why the result was getting truncated to length 21, I noticed we were doing something similar to this patch on lines 756 and 778.

Reply:
It looks like we changing the parameter sent by MDB to us. This might be dangerous thus we need to change the length of the expected result in MDB code similar to what had done in MDEV-27320.